azerothcore/azerothcore-wotlk
# fix(DB/loot): Fix drops related to Root Samples Q

I launched the worldserver with the `Calculate.Gameoject.Zone.Area.Data` setting enabled to fill out the `areaId` and `zoneId` columns in the `gameobject` table. I collected the guids of all related herb nodes in the Barrens zone using this information. The guids are hardcoded into the sql because the `areaId` and `zoneId` stuff is not available during the DB assembly.



## Changes Proposed:
- Make all related herb nodes in the Barrens zone be the 'special' variants
- Make all related herb nodes outside the Barrens zone be the 'normal' variants
- Adjust loot tables 

## Issues Addressed:

- Closes #5792

## SOURCE:

https://classic.wowhead.com/

## Tests Performed:

- Walked around inside and outside the Barrens zone and inspected herb nodes
- Ran the test queries described below

## How to Test the Changes:


1. Check some nodes in-game if you want to
2. Run these queries:

Loot tables:
```sql
CREATE TEMPORARY TABLE `herbs` (
    `name` VARCHAR(20)
);

INSERT INTO `herbs` VALUES
('Silverleaf'),
('Earthroot'),
('Peacebloom'),
('Mageroyal'),
('Briarthorn'),
('Bruiseweed');

SELECT DISTINCT
    gobt.entry AS gobt_id, 
    gobt.name AS gobt, 
    it.entry AS item_id,
    it.name AS item, 
    chance AS '%', 
    questrequired AS q, 
    goblt.mincount AS min, 
    goblt.maxcount AS max
FROM 
    `gameobject_template` gobt 
    INNER JOIN `herbs` h ON gobt.name = h.name COLLATE utf8mb4_general_ci
    INNER JOIN `gameobject_loot_template` goblt ON gobt.data1 = goblt.entry
    INNER JOIN `item_template` it ON goblt.item = it.entry
ORDER BY gobt_id ASC;
```
(the `COLLATE ...` part has to do with character encoding and you might not need to include it)

Expected results:
```
+---------+------------+---------+--------------+-----+---+-----+-----+
| gobt_id | gobt       | item_id | item         | %   | q | min | max |
+---------+------------+---------+--------------+-----+---+-----+-----+
|    1617 | Silverleaf |     765 | Silverleaf   | 100 | 0 |   1 |   3 |
|    1618 | Peacebloom |    2447 | Peacebloom   | 100 | 0 |   1 |   3 |
|    1619 | Earthroot  |    2449 | Earthroot    | 100 | 0 |   1 |   3 |
|    1620 | Mageroyal  |    2452 | Swiftthistle |  20 | 0 |   1 |   2 |
|    1620 | Mageroyal  |     785 | Mageroyal    | 100 | 0 |   1 |   3 |
|    1621 | Briarthorn |    2450 | Briarthorn   | 100 | 0 |   1 |   3 |
|    1621 | Briarthorn |    2452 | Swiftthistle |  40 | 0 |   1 |   2 |
|    1622 | Bruiseweed |    2453 | Bruiseweed   | 100 | 0 |   1 |   3 |
|    3724 | Peacebloom |    5056 | Root Sample  | 100 | 1 |   1 |   1 |
|    3724 | Peacebloom |    2447 | Peacebloom   | 100 | 0 |   1 |   3 |
|    3725 | Silverleaf |    5056 | Root Sample  | 100 | 1 |   1 |   1 |
|    3725 | Silverleaf |     765 | Silverleaf   | 100 | 0 |   1 |   3 |
|    3726 | Earthroot  |    5056 | Root Sample  | 100 | 1 |   1 |   1 |
|    3726 | Earthroot  |    2449 | Earthroot    | 100 | 0 |   1 |   3 |
|    3727 | Mageroyal  |     785 | Mageroyal    | 100 | 0 |   1 |   3 |
|    3727 | Mageroyal  |    5056 | Root Sample  | 100 | 1 |   1 |   1 |
|    3727 | Mageroyal  |    2452 | Swiftthistle |  20 | 0 |   1 |   2 |
|    3729 | Briarthorn |    2450 | Briarthorn   | 100 | 0 |   1 |   3 |
|    3729 | Briarthorn |    5056 | Root Sample  | 100 | 1 |   1 |   1 |
|    3729 | Briarthorn |    2452 | Swiftthistle |  40 | 0 |   1 |   2 |
|    3730 | Bruiseweed |    5056 | Root Sample  | 100 | 1 |   1 |   1 |
|    3730 | Bruiseweed |    2453 | Bruiseweed   | 100 | 0 |   1 |   3 |
+---------+------------+---------+--------------+-----+---+-----+-----+
```

Count of 'normal' herb nodes inside the Barrens:
```sql
SELECT COUNT(*) FROM `gameobject` WHERE ID BETWEEN 1617 and 1622 AND (`zoneId` = 17 OR `areaId` = 17);
```
Expected result: 0

Count of 'special' herb nodes outside the Barrens:
```sql
SELECT COUNT(*) FROM `gameobject` WHERE ID IN (3724, 3725, 3726, 3727, 3729, 3730) AND NOT (`zoneId` = 17 OR `areaId` = 17);
```
Expected result: 0

The last two queries access the `areaId` and `zoneId` columns, so you will also need to run the worldserver with `Calculate.Gameoject.Zone.Area.Data` enabled before running them, otherwise you'll get nonsensical results.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
